smoke-tests/images/servlet/servlet-3.0/src/main/java/io/opentelemetry/smoketest/matrix/AsyncGreetingServlet.java

```diff
@@ -56,14 +58,15 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
   }
 
   private static void handleRequest(AsyncContext ac) {
-    System.err.println("dispatch async request");
+    System.err.println("handle async request");
     try {
-      ac.dispatch("/greeting");
-      System.err.println("async request dispatched");
+      greetingServlet.doGet(
+          (HttpServletRequest) ac.getRequest(), (HttpServletResponse) ac.getResponse());
+      ac.complete();
+      System.err.println("async request handled");
     } catch (Throwable throwable) {
-      System.err.println("dispatching async request failed");
+      System.err.println("handling async request failed");
       throwable.printStackTrace();
-      throw throwable;
```

Comment:
why not rethrow?

Reply:
It doesn't really matter whether it is rethrown or not, this code is running in the executor we created and the exception won't reach the servlet container anyway. If there is an exception the test will fail with timeout regardless of whether it is rethrown or not.